security/s2a/internal/crypter/ciphersuite.go

```diff
@@ -0,0 +1,109 @@
+package crypter
+
+import (
+	"crypto/sha256"
+	"golang.org/x/crypto/sha3"
+	s2a_proto "google.golang.org/grpc/security/s2a/internal/proto"
+	"hash"
+)
+
+// ciphersuite is the interface for a ciphersuite used in the S2A Half
+// Connection.
+type ciphersuite interface {
+	// keySize returns the key size in bytes.
+	keySize() int
+	// nonceSize returns the nonce size in bytes.
+	nonceSize() int
+	// trafficSecretSize returns the traffic secret size in bytes.
+	trafficSecretSize() int
+	// hashFunction returns the hash function for the ciphersuite.
+	hashFunction() func() hash.Hash
```

Comment:
WDYT of adding hash digest length to the interface? We'll need this for the record protocol.

Reply:
Whoops sorry - this is the same as trafficSecretSize.